pkg/operation/cloudbotanist/awsbotanist/machines.go

```diff
@@ -77,6 +78,9 @@ func (b *AWSBotanist) GenerateMachineConfig() ([]map[string]interface{}, operati
 		return nil, nil, err
 	}
 
+	utils.ConsistentRandomSort(b.Shoot.Info.Namespace, b.Shoot.Info.Name, len(zones), func(i, j int) {
```

Comment:
This change is not backwards compatible because `"subnetID":         stateVariables[tfOutputNameSubnet(zoneIndex)],` would return another ID not matching the subnet in the correct zone after the random sort.

Do we really need/want this? Actually, it doesn't hurt if the last zone gets one VM more than the others in case the replica count is uneven. Anyway, if you can't fix it easily please file another pull request.